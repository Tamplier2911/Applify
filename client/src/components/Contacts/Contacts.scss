/*
.contacts {
  &__header {
    font-family: var(--font-title);
    font-size: 4rem;
    font-weight: 300;
    padding: 1rem;

    @media only screen and (max-width: 768px) {
      line-height: 1;
    }
  }

  &__formComposition {
    position: relative;
    display: grid;
    grid-template-columns: repeat(10, 1fr);
    grid-template-rows: repeat(10, 1fr);
    padding: 4rem;

    @media only screen and (max-width: 650px) {
      padding: 2rem;
    }

    @media only screen and (max-width: 450px) {
      padding: 0rem;
    }

    &--left {
      position: relative;
      grid-area: 2 / 1 / 10 / 5;
      z-index: 5;

      @media only screen and (max-width: 1000px) {
        grid-area: 1 / 4 / 6 / 8;
      }

      @media only screen and (max-width: 900px) {
        grid-area: 1 / 3 / 6 / 8;
      }

      @media only screen and (max-width: 650px) {
        grid-area: 1 / 4 / 6 / 8;
      }

      @media only screen and (max-width: 550px) {
        grid-area: 1 / 3 / 6 / 8;
      }

      @media only screen and (max-width: 425px) {
        grid-area: 1 / 3 / 6 / -1;
      }

      @media only screen and (max-width: 360px) {
        grid-area: 1 / 2 / 6 / -1;
      }

      @media only screen and (max-width: 310px) {
        grid-area: 1 / 1 / 4 / -1;
      }

      &--svg {
        height: 40rem;

        @media only screen and (max-width: 1000px) {
          height: 30rem;
        }

        @media only screen and (max-width: 650px) {
          height: 20rem;
        }

        @media only screen and (max-width: 250px) {
          height: 10rem;
        }
      }
    }

    &--right {
      grid-area: 1 / 2 / -1 / -1;
      display: grid;
      grid-template-columns: repeat(2, 1fr);

      background-color: var(--cl-contacts-bg);
      box-shadow: 0 0.2rem 0.5rem var(--cl-square-shadow);

      @media only screen and (max-width: 1000px) {
        grid-template-columns: 1fr;
        grid-template-rows: repeat(3, 1fr);
        grid-row-gap: 8rem;
        grid-area: 2 / 1 / -1 / -1;
      }

      @media only screen and (max-width: 650px) {
        grid-row-gap: 2rem;
      }

      @media only screen and (max-width: 650px) {
        grid-template-rows: repeat(10, 1fr);
      }

      @media only screen and (max-width: 425px) {
        grid-template-rows: repeat(5, 1fr);
        grid-row-gap: 10rem;
      }

      @media only screen and (max-width: 245px) {
        grid-template-rows: repeat(10, 1fr);
        grid-row-gap: 0rem;
      }

      &--formContainer {
        grid-column: 2 / -1;
        display: grid;
        //   grid-template-rows: min-content;
        //   grid-auto-rows: min-content;
        padding: 2rem;

        @media only screen and (max-width: 1000px) {
          grid-column: 1 / -1;
          grid-row: 2 / -1;
        }

        @media only screen and (max-width: 650px) {
          grid-row: 4 / -1;
        }

        @media only screen and (max-width: 425px) {
          grid-row: 2 / -1;
        }
      }
    }
  }
}

// 590 - 425

*/
