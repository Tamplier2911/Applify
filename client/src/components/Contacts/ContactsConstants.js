const contactsData = {
  eng: {
    contactsHeader:
      "Ready to enhance your business with cutting edge technologies?",
  },
  rus: {
    contactsHeader:
      "Готовы расширить свой бизнес с использованием новейших технологий?",
  },
  ukr: {
    contactsHeader:
      "Готові розширити свій бізнес з використанням новітніх технологій?",
  },
};

export default contactsData;
