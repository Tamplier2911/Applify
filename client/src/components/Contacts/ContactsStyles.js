import styled, { css } from "styled-components";

// svg
import { ReactComponent as DevSVG } from "../../assets/svg/web_developer.svg";

const getFont = props => {
  const { lang } = props;
  if (lang === "eng") {
    return `font-family: var(--font-title);`;
  } else if (lang === "rus" || lang === "ukr") {
    return `font-family: var(--font-cyrillic);`;
  }
  return `font-family: var(--font-title);`;
};

const fadeContactsIn = css`
  @keyframes fadeContactsIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`;

export const ContactsContainer = styled.div`
  ${fadeContactsIn}
  animation: 1s fadeContactsIn;
`;

export const ContactsTitle = styled.h2`
  ${getFont}
  font-size: 4rem;
  font-weight: 300;
  padding: 1rem;

  @media only screen and (max-width: 768px) {
    line-height: 1;
  }
`;

export const ContactsFormComp = styled.div`
  position: relative;
  display: grid;
  grid-template-columns: repeat(10, 1fr);
  grid-template-rows: repeat(10, 1fr);
  padding: 4rem;

  @media only screen and (max-width: 650px) {
    padding: 2rem;
  }

  @media only screen and (max-width: 450px) {
    padding: 0rem;
  }
`;

export const ContactsFormCompLeft = styled.div`
  position: relative;
  grid-area: 2 / 1 / 10 / 5;
  z-index: 5;

  @media only screen and (max-width: 1000px) {
    grid-area: 1 / 4 / 6 / 8;
  }

  @media only screen and (max-width: 900px) {
    grid-area: 1 / 3 / 6 / 8;
  }

  @media only screen and (max-width: 650px) {
    grid-area: 1 / 4 / 6 / 8;
  }

  @media only screen and (max-width: 550px) {
    grid-area: 1 / 3 / 6 / 8;
  }

  @media only screen and (max-width: 425px) {
    grid-area: 1 / 3 / 6 / -1;
  }

  @media only screen and (max-width: 360px) {
    grid-area: 1 / 2 / 6 / -1;
  }

  @media only screen and (max-width: 310px) {
    grid-area: 1 / 1 / 4 / -1;
  }
`;

export const ContactsFormCompSVG = styled(DevSVG)`
  height: 40rem;

  @media only screen and (max-width: 1000px) {
    height: 30rem;
  }

  @media only screen and (max-width: 650px) {
    height: 20rem;
  }

  @media only screen and (max-width: 245px) {
    height: 10rem;
  }
`;

export const ContactsFormCompRight = styled.div`
  grid-area: 1 / 2 / -1 / -1;
  display: grid;
  grid-template-columns: repeat(2, 1fr);

  background-color: var(--cl-contacts-bg);
  box-shadow: 0 0.2rem 0.5rem var(--cl-square-shadow);

  @media only screen and (max-width: 1000px) {
    grid-template-columns: 1fr;
    grid-template-rows: repeat(3, 1fr);
    grid-row-gap: 8rem;
    grid-area: 2 / 1 / -1 / -1;
  }

  @media only screen and (max-width: 650px) {
    grid-row-gap: 2rem;
  }

  @media only screen and (max-width: 650px) {
    grid-template-rows: repeat(10, 1fr);
  }

  @media only screen and (max-width: 425px) {
    grid-template-rows: repeat(5, 1fr);
    grid-row-gap: 10rem;
  }

  @media only screen and (max-width: 245px) {
    grid-template-rows: repeat(10, 1fr);
    grid-row-gap: 0rem;
  }
`;

export const ContactsFormCompFormWrap = styled.div`
  grid-column: 2 / -1;
  display: grid;
  padding: 2rem;

  @media only screen and (max-width: 1000px) {
    grid-column: 1 / -1;
    grid-row: 2 / -1;
  }

  @media only screen and (max-width: 650px) {
    grid-row: 4 / -1;
  }

  @media only screen and (max-width: 425px) {
    grid-row: 2 / -1;
  }
`;
