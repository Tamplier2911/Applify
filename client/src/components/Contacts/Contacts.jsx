// import "./Contacts.scss";
import React from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";

// components
import FormHolder from "../FormHolder/FormHolder";
import ContactsForm from "../ContactsForm/ContactsForm";

// JS Rendering CSS
import { ContactsContainer } from "./ContactsStyles";

// component constants
import contactsData from "../../utils/ComponentContactsConstants/componentContactsConstants";

const Contacts = ({ lang }) => {
  const { contactsHeader } = contactsData[lang];
  return (
    <ContactsContainer>
      <FormHolder type={"contact"} title={contactsHeader}>
        <ContactsForm />
      </FormHolder>
    </ContactsContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage
});

export default connect(mapStateToProps)(Contacts);
