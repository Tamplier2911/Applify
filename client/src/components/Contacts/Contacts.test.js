import React from "react";
import toJson from "enzyme-to-json";

// component
import Contacts from "./Contacts";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<Contacts />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<Contacts store={store} />);
  });

  it("renders without an error", () => {
    const contacts = findByTestAttr(wrapper, "contacts");
    expect(contacts.length).toBe(1);
  });

  it("renders children - form holder", () => {
    const contactsFormHolder = findByTestAttr(wrapper, "contacts-form-holder");
    expect(contactsFormHolder.length).toBe(1);
  });

  it("matches snapshot", () => {
    const contacts = findByTestAttr(wrapper, "contacts");
    expect(toJson(contacts)).toMatchSnapshot();
  });
});
