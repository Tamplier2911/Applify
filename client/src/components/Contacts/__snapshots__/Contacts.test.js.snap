// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contacts /> matches snapshot 1`] = `
<styled.div
  data-test="contacts"
>
  <FormHolder
    data-test="contacts-form-holder"
    title="Ready to enhance your business with cutting edge technologies?"
    type="contact"
  >
    <Connect(ContactsForm) />
  </FormHolder>
</styled.div>
`;
